packages/manager/modules/hub/src/components/payment-status-tile/AutorenewService.class.js

```diff
@@ -0,0 +1,81 @@
+import includes from 'lodash/includes';
+import snakeCase from 'lodash/snakeCase';
+
+import { DEBT_STATUS } from './payment-status-tile.constants';
+
+export default class AutorenewService {
+  constructor(service) {
+    Object.assign(this, service);
+  }
+
+  getRenew() {
+    if (this.hasManualRenew()) {
+      return 'manualPayment';
+    }
+
+    if (this.shouldDeleteAtExpiration() && !this.isResiliated()) {
+      return 'delete_at_expiration';
+    }
+
+    if (this.isResiliated()) {
+      return 'expired';
+    }
+
+    if (this.hasAutomaticRenew() || this.hasForcedRenew()) {
+      return 'automatic';
+    }
+
+    return 'manualPayment';
+  }
+
+  hasAutomaticRenew() {
+    return this.renew.automatic;
+  }
+
+  hasAutomaticRenewal() {
+    return this.hasForcedRenew() || this.hasAutomaticRenew();
+  }
+
+  hasDebt() {
+    return includes(DEBT_STATUS, snakeCase(this.status).toUpperCase());
```

Comment:
again if possible, turn this into instance variable, because it involves looping array, angular call this multiple times during digest cycle

Reply:
This is fixed :)